tests/core/providers/test_http_provider.py

```diff
@@ -127,26 +127,9 @@ def test_http_empty_batch_response(mock_post):
     )
     w3 = Web3(HTTPProvider())
     with w3.batch_requests() as batch:
+        assert w3.provider._is_batching
         with pytest.raises(Web3RPCError, match="empty batch"):
             batch.execute()
 
     # assert that even though there was an error, we have reset the batching state
     assert not w3.provider._is_batching
-
-
-@patch(
-    "web3._utils.http_session_manager.HTTPSessionManager.make_post_request",
-    new_callable=Mock,
-)
-def test_sync_provider_is_batching_when_make_batch_request(mock_post):
```

Comment:
@pacrob I'm going to put all these back. The original implementation for thread safety didn't lend itself well to this test anymore but I put back `_is_batching` as the `ContextVar` holder / flag so I will put these back now!